### PR TITLE
Makes build cake 1.0.0 compatible

### PR DESCRIPTION
**Pull request by abriening**
Fixes / New Feature #

## Proposed Changes

  - Adds .config with dotnet-tools to load cake.tool
  - restore and use `dotnet cake` in Makefile
  - Fixes change in `DeleteDirectory` signature

### CODE REVIEW

**Review by amamei — APPROVED** [review]

**Review by gcaraman — APPROVED** [review]

**Review by CristaliS — APPROVED** [review]

**Comment by raman-m** [comment]
Please do the following:
- Merge [PR 1](https://github.com/abriening/Ocelot/pull/1)
- Rebase feature branch onto develop
- Resolve all merge conflicts

**Comment by raman-m** [comment]
@abriening Hi Andrew!
I've tried to solve and resolve merge conflicts in better manner.
So, the [feature branch](/abriening/Ocelot/tree/cake-1-0-0-compatible) has been [rebased](#event-10172512646) onto [ThreeMammals:develop](../tree/develop) successfully!
I've kept all changes from target branch because it contains most actual config for builds.
But now, I don't see any changes in your PR at all. Small change on [line 444](../compare/develop...abriening:Ocelot:cake-1-0-0-compatible#diff-c6eace5a70f0c2d920dfa97e988d4cad3d0131dd2b029bb2646c96222e866cf8R444). Is it relevant?
Also, Cake Tool has version [3.0.0](../blob/develop/.config/dotnet-tools.json#L6) now, not **1.0.0** as it's requested in PR.
**It seems your PR is obsolete now, right?**

**Comment by abriening** [comment]
My PRs were ignored for several years. I've moved to https://github.com/microsoft/reverse-proxy instead of ocelot.

**Comment by raman-m** [comment]
Andrew!
Sorry for ignoring your PR. The owner doesn't care about this project since 2020.
Can we close the PR because of applied fix by owner in other commits?

**Review by raman-m — CHANGES_REQUESTED** [review]

File: build.cake [hunk 1]
```diff
@@ -441,22 +441,22 @@ private void PublishPackages(ConvertableDirectoryPath packagesDir, ConvertableFi
 
 private void CreateGitHubRelease()
 {
-	var json = $"{{ \"tag_name\": \"{versioning.NuGetVersion}\", \"target_commitish\": \"main\", \"name\": \"{versioning.NuGetVersion}\", \"body\": \"{ReleaseNotesAsJson()}\", \"draft\": true, \"prerelease\": true }}";
-	
+	var json = $"{{ \"tag_name\": \"{versioning.NuGetVersion}\", \"target_commitish\": \"master\", \"name\": \"{versioning.NuGetVersion}\", \"body\": \"{ReleaseNotesAsJson()}\", \"draft\": true, \"prerelease\": true }}";
```
Comment (raman-m) [inline build.cake:444]:
```suggestion
	var json = $"{{ \"tag_name\": \"{versioning.NuGetVersion}\", \"target_commitish\": \"main\", \"name\": \"{versioning.NuGetVersion}\", \"body\": \"{ReleaseNotesAsJson()}\", \"draft\": true, \"prerelease\": true }}";
```
We make releases on **main** which protected!
**master** is not protected anymore, and [this branch was removed](../branches/all).

**Review by raman-m — CHANGES_REQUESTED** [review]

**Pull request closed without merging**